crates/exex/exex/src/notifications.rs

```diff
@@ -21,6 +21,45 @@ pub struct ExExNotifications<P, E> {
     inner: ExExNotificationsInner<P, E>,
 }
 
+/// A trait, that represents a stream of [`ExExNotification`]s. The stream will emit notifications
+/// for all blocks. If the stream is configured with a head via [`ExExNotifications::set_with_head`]
+/// or [`ExExNotifications::with_head`], it will run backfill jobs to catch up to the node head.
+pub trait ExExNotificationsStream<P, E>:
+    Stream<Item = eyre::Result<ExExNotification>> + Unpin
+where
+    P: BlockReader + HeaderProvider + StateProviderFactory + Clone + Unpin + 'static,
+    E: BlockExecutorProvider + Clone + Unpin + 'static,
+{
```

Comment:
do we even need these generics here?

they're not used in the trait itself, this more like a StreamExt kinda trait

Reply:
it can be just `Unpin` but it seems not clear, isn't it?

Reply:
oh, that's a great point, because the whole idea behind abstracting the `ExExNotifications` behind the trait is that we can avoid generics. We should be able to remove the `E, P` altogether.

Reply:
I see, that can be removed, agreed, as they are already on `ExExNotifications`